game_board.gd

```diff
@@ -294,12 +313,14 @@ func _move_active_unit(new_cell: Vector2) -> void:
 	await _active_unit.walk_finished
 	_reinitialize()
 	# Finally, we clear the `_active_unit`, which also clears the `_walkable_cells` array.
+	#if _active_unit.actionPoints <= 0:
```

Comment:
need to get rid of this check, it is wrong and causes a bug and needs to be taken out back and deleted